src/SourceLocator/Type/DirectorySourceLocator.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace BetterReflection\SourceLocator\Type;
+
+use BetterReflection\Identifier\Identifier;
+use BetterReflection\Identifier\IdentifierType;
+use BetterReflection\Reflector\Reflector;
+use BetterReflection\SourceLocator\Exception\InvalidDirectory;
+
+/**
+ * This source locator loads all php files in an entire directories.
+ */
+class DirectorySourceLocator implements SourceLocator
+{
+
+    /**
+     * @var AggregateSourceLocator
+     */
+    private $aggregatedSourceLocator;
+
+    /**
+     * @param $directories array directories to scan
+     */
+    public function __construct(array $directories)
+    {
+        $sourceLocators = [];
+        foreach ($directories as $dir) {
+            $dir = (string) $dir;
+            if (!is_dir($dir)) {
+                throw new InvalidDirectory(sprintf('Is not a directory: %s', $dir));
```

Comment:
@Ocramius FYI thinking of dropping PHP 5.6 after the next minor (may result in a major release if there's BC breaks elsewhere anyway), so yeah, we'll get better here soon